traitsui/editor.py

```diff
@@ -447,7 +447,7 @@ def _sync_values(self):
             value of corresponding factory traits.
         """
         factory = self.factory
-        for name, trait in factory.traits(sync_value=not_none):
+        for name, trait in factory.traits(sync_value=not_none).items():
```

Comment:
Gah. How on earth was this working before?

Reply:
It wasn't.  Nothing was using it.